Update docs of @woocommerce/api (#32004)
woocommerce/woocommerce: packages/js/api/README.md

# WooCommerce API Client

> An API client for interacting with WooCommerce installations that works both on Browser and Node environments. Here are the current and planned

Reviewer: Minor wording change suggestion:

```suggestion
An API client for interacting with WooCommerce installations that works both on in the browser and in Node environments. Here are the current and planned
```

Author: Thanks, commited this suggestion separatedly as there as a small typo in the suggested commit https://github.com/woocommerce/woocommerce/pull/32004/commits/922a72ef7e18a1faac9a408a0855e9f16fc78b01. The typo was "that works both **on** in the browser"

features:

- [x] TypeScript Definitions \*
- [x] Axios API Client with support for OAuth & basic auth
- [X] Partial support to Repositories, to simplify interaction with basic data types \*
- [x] Service classes for common activities such as changing settings

> _\* TypeScript Definitions and Repositories are only supported for Products, and partially supported for Orders._

Reviewer: Suggestion on linking to the API documentation, let me know what you think:

```suggestion
_\* TypeScript Definitions and Repositories are only supported for [Products](https://woocommerce.github.io/woocommerce-rest-api-docs/#products), and partially supported for [Orders](https://woocommerce.github.io/woocommerce-rest-api-docs/#orders)._
```

Alternatively, we can look at linking to the Repositories in the code itself instead.

Author: Commited, thank you!

> _\* TypeScript Definitions and Repositories are only supported for Products, and partially supported for Orders._

Reviewer: Also might add the word `currently` just to show this is actively being worked on:

```suggestion
_\* TypeScript Definitions and Repositories are currently only supported for Products, and partially supported for Orders._
```

Author: Comitted, thank you. https://github.com/woocommerce/woocommerce/pull/32004/commits/4cfb412a57aeb6ebe5aa601d50621c450e782be5

## Differences from @woocommerce/woocomerce-rest-api

> WooCommerce has two API clients in JavaScript for interacting with WooCommerce installation. This one, and [@woocommerce/woocomerce-rest-api](https://www.npmjs.com/package/@woocommerce/woocommerce-rest-api).

Reviewer: ```suggestion
WooCommerce has two API clients in JavaScript for interacting with a WooCommerce installation's RESTful API. This package, and the [@woocommerce/woocomerce-rest-api](https://www.npmjs.com/package/@woocommerce/woocommerce-rest-api) package.
```

Author: Commited, thank you.

> The main difference between them is the Repositories and the TypeScript definitions for the supported endpoints. When using Axios directly, as you can do with both libraries, you query the WooCommerce API in a raw object format, following the API documentation parameters.  With Repositories, you have the parameters as properties of an object, so you have auto-complete and strict types, for instance.

Reviewer: Let's link to the API docs here:

```suggestion
The main difference between them is the Repositories and the TypeScript definitions for the supported endpoints. When using Axios directly, as you can do with both libraries, you query the WooCommerce API in a raw object format, following the [API documentation](https://woocommerce.github.io/woocommerce-rest-api-docs/#introduction) parameters.  With Repositories, you have the parameters as properties of an object, so you have auto-complete and strict types, for instance.
```

I also think it may be worth clarifying here instead of saying "With Repositories" we also note that it's the Repositories within this package, so something like:

> Comparatively, with the Repositories provided in this package, you have the parameters as properties of an object, which gives you the benefits of auto-complete and strict types, for instance.

Author: Commited, thank you. https://github.com/woocommerce/woocommerce/pull/32004/commits/d45f7051a4d0f99bdf81980f952a6d02f392e453
